### PR TITLE
Update changelog for the Azurite fix

### PR DESCRIPTION
**Pull request by antkmsft**
Update changelog for the issue #4206 that was fixed in PR #4208. Thank you @ahojnnes!

### CODE REVIEW

**Review by ahsonkhan — APPROVED** [review]
Thanks!

File: sdk/core/azure-core/CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,14 @@
 
 ### Bugs Fixed
 
+- [[#4206]](https://github.com/Azure/azure-sdk-for-cpp/issues/4206) Azurite parallel connectivity problem.
```
Comment (LarryOsterman) [inline sdk/core/azure-core/CHANGELOG.md:11]: From what I understand, this is not an issue with Azurite, but instead a general problem which occurs whenever a server terminates the TCP connection during the HTTP exchange.

Comment (LarryOsterman) [reply sdk/core/azure-core/CHANGELOG.md:11]: Maybe "Fixed connectivity issues which can occur if a TCP connection is dropped prematurely"?

Comment (antkmsft) [reply sdk/core/azure-core/CHANGELOG.md:11]: @LarryOsterman, #4262

**Pull request merged**